subsys/bluetooth/controller/Kconfig

```diff
@@ -239,14 +248,16 @@ if BT_CONN
 
 config BT_CTLR_LE_ENC
```

Comment:
Maybe we should have an easier way for controllers to determine support for the different features.

How about
config BT_CTRL_LE_SUPPORT
  bool

config BT_CTLR_LE_ENC
  bool "LE Encryption
  depends on BT_CTRL_LE_ENC_SUPPORT
  default y

config BT_LLL_VENDOR_NORDIC
	bool "Use Nordic LLL"
        select BT_LL_ENC_SUPPORT

Reply:
Right, so all the various options would get set under the corresponding BT_LL_VENDOR_*, making things much more readable, instead of being scattered across all over the place (as I've done it). @cvinayak , what do you think? Should I go ahead and update the Kconfigs accordingly?

Reply:
Agree. Yes, please go ahead.

Reply:
@alexandru-porosanu-nxp This PR is quite huge, if you prefer we can take the suggested change as separate PR afterwards.

Reply:
Actually, I went ahead and done it anyway. It should be there in the latest version of the PR. So @joerchan , if you could have a look and let me know if it looks OK, that would be great.

Reply:
Thanks. This looks much more managable. The last 3 features I think are very NRF specific. And I think maybe should be moved under LLL_VENDOR_NORDIC as they are closely related to NRF hardware features.